### DISCUSSION TITLE
Can I refine and add a dynamic path?

### DISCUSSION THREAD

**Discussion by sndrem** [colinhacks/zod#2501] [Q&A]
I have a list of dynamic fields which is being rendered using the `useFieldArray`-hook from react-form-hook. I have a zod schema which calls refine and looks like this 

```js
.refine(
        (data) => {
          return data.filter((d) => d.navIdent !== "").length > 0;
        },
        {
          message: "Du må velge minst en kontaktperson i NAV",
          path: ["navIdent"],
        }
      )
```
The `path` property does not work in this example, but it works if I hard code `path: ["0.navIdent"]`. Is it possible to have a wildcard for paths, or are there any other solutions on how to dynamically map to the correct path?

**Comment by feJustino** [comment]
I'm dealing with the same thing.
The way I managed to solve this problem was by including the error in the root of the array, and using the error from the root array's state to populate the error. It solved my problem in a simple way.

```jsx 
    responsibles: z
      .array(ResponsibleSchema)
      .min(1, REQUIRED_MESSAGE)
      .refine(
        (responsibles) => {
          const total = responsibles.reduce(
            (sum, resp) => sum + Number(resp.quantityQuestion),
            0
          )
          return total <= 100
        },
        {
          message:
            'O total de questões é de até 100 por simulado. Ajuste a quantidade para continuar.',
        }
      ),
  })
```


```jsx

        error={
          responsibleErros?.[index]?.quantityQuestion?.message ||
          responsibleErros?.root
        }
        
```
